benchmarks/src/models/architectures/yolo/__init__.py

```diff
@@ -0,0 +1,113 @@
+        if org_tensor.shape != new_tensor.shape:
+            print("weight file has incorrect shape for this tensor")
+            return
+        if x.shape[0] == 0:
+            print("No more data!")
+            return
+        x = torch.from_numpy(x).view_as(new_tensor)
+        with torch.no_grad():
+            new_tensor.copy_(x)
+
+    for idx, (old_module, new_module) in enumerate(
+        zip(network_for_weight_file, new_model.network)
+    ):
+        assert isinstance(old_module, type(new_module))
+        # Weights are loaded only to convolutional layers
+        if not isinstance(new_module, nn.Sequential):
+            continue
+
+        print(f"Loading weight for {idx}")
+        new_conv = new_module[0]
+        old_conv = old_module[0]
+        assert isinstance(new_conv, nn.Conv2d)
+        assert isinstance(old_conv, nn.Conv2d)
+
+        # Convolution may be followed by batch normalization, in which case we read the batch
+        # normalization parameters and not the convolution bias.
+        if len(new_module) > 1 and isinstance(new_module[1], nn.BatchNorm2d):
+            assert isinstance(old_module[1], nn.BatchNorm2d)
+            new_bn = new_module[1]
+            old_bn = old_module[1]
+            read(old_bn.bias, new_bn.bias)
+            read(old_bn.weight, new_bn.weight)
+            read(old_bn.running_mean, new_bn.running_mean)
+            read(old_bn.running_var, new_bn.running_var)
+        else:
+            read(old_conv.bias, new_conv.bias)
+
+        read(old_conv.weight, new_conv.weight)
+
+
+def load_darknet_weights(weight_file, model):
+    default_model_lines = generate_config(DEFAULT_CLASSES)
+    default_model_network = YOLOConfiguration(default_model_lines).get_network()
+    _load_darknet_weights(default_model_network, model, weight_file)
+
+
+def initialize_yolo(
+    # YOLO does not consider the background to be a unique class
+    num_classes: int = NUM_OUTPUT_CLASSES - 1,
+    pretrained: bool = True,
+    # https://github.com/AlexeyAB/darknet/issues/4983
+    confidence_threshold: float = 0.005,
+    nms_threshold: float = 0.7,
+    max_predictions_per_image: int = -1,
+) -> Tuple[nn.Module, float]:
+
+    config_lines = generate_config(num_classes)
+    model_config = YOLOConfiguration(config_lines)
+    model = YOLO(
+        network=model_config.get_network(),
+        confidence_threshold=confidence_threshold,
+        nms_threshold=nms_threshold,
+        max_predictions_per_image=max_predictions_per_image,
+        image_height=model_config.global_config["height"],
+        image_width=model_config.global_config["width"],
+    )
+    if pretrained:
+        if not DEFAULT_WEIGHT_PATH.exists():
+            download_from_url(YOLOv4_DARKNET_WEIGHTS, DEFAULT_WEIGHT_PATH)
+        with DEFAULT_WEIGHT_PATH.open() as weight_file:
+            load_darknet_weights(weight_file, model)
+    return model, 0.001
```

Comment:
Instead of returning the default LR as a hard-coded value at the bottom. Could we have it as input to the initialize yolo for example? And set the default value there? I think it's a bit ambiguous left as is

Reply:
Or somewhere else if that doesn't work

Reply:
The user has an option to override the default in the [model class](https://github.com/SmallRobotCompany/smallssd/blob/d2eea82818a0c2ccdcebb750c81a20443707fbcf/benchmarks/src/models/base.py#L16).

I've also created a `DEFAULT_YOLO_LR` variable to make it more explicit what's being returned

Reply:
Looks good to me!